lib/ecto_sharding/query_processing.ex

```diff
@@ -30,7 +30,7 @@ defmodule EctoSharding.QueryProcessing do
           schema -> schema
         end
 
-      if model.sharded? do
+      if Keyword.has_key?(model.__info__(:functions), :sharded?) && model.sharded? do
```

Comment:
why did this get added again? would it be worth pulling this change into a separate PR to keep concerns separate here, or was this specific to the migrations work?

Reply:
This makes migrations on the main db possible because when Ecto tries to read the schema_migrations it fails because it uses Ecto.Repo.

Reply:
Sorry, wasn't Ecto.Repo....was the fact that there is no Schema defined in the project.  Ecto builds that on the fly and there's no opportunity to make it inherit from EctoSharding.Schema, so the sharded? method doesn't exist and blows up.  Changing this allows it to work when the actual 'model' is an EctoSharding.Schema, but the schema migrations is not.  I don't think there is any equivalent of a `try` method in Elixir, is there?  That's effectively what this is doing.